app/models/mixins/purging_mixin.rb

```diff
@@ -46,6 +46,26 @@ def purge_by_date(older_than, window = nil, &block)
       total
     end
 
+    def purge_count_by_remaining(remaining)
+      purge_ids_for_remaining(remaining).size
+    end
+
+    def purge_by_remaining(remaining, window = nil, &block)
+      _log.info("Purging #{table_name.humanize} older than last #{remaining} results...")
+      total = purge_in_batches(purge_ids_for_remaining(remaining), window || purge_window_size, &block)
+      _log.info("Purging #{table_name.humanize} older than last #{remaining} results...Complete - Deleted #{total} records")
+      total
+    end
+
+    def purge_ids_for_remaining(remaining)
```

Comment:
Would this method documentation, or something like it, be a little more descriptive?

```ruby
# Private:  Find the ids to purge based on a fixed number to remain for each
# resource (remaining)
# 
# Assigns a "RANK" value to each of the returned records, categorized by their
# foreign key(s) (for `DriftState`, it is `resource_type, resource_id`, for
# `MiqReportResult`, it is `miq_report_result_id`).  The values to purge are
# then the ones whose rank is greater than the number to keep.
# 
# So given the following DriftStates:
# 
# {id: 1, resource_type: 'foo', resource_id: 1}  # Rank = 1
# {id: 2, resource_type: 'foo', resource_id: 1}  # Rank = 2
# {id: 3, resource_type: 'foo', resource_id: 2}  # Rank = 1
# {id: 4, resource_type: 'foo', resource_id: 2}  # Rank = 2
# {id: 5, resource_type: 'bar', resource_id: 6}  # Rank = 1
# {id: 6, resource_type: 'baz', resource_id: 4}  # Rank = 1
# 
# For a "remaining" value of `1`, id 2 and 4 would then be deleted.
def purge_ids_for_remaining(remaining)
```